packages/app/src/components/inspector/ColorInputFieldWithPicker.tsx

```diff
@@ -150,12 +152,15 @@ export default memo(function ColorInputFieldWithPicker({
   // inspector rows may also take arrays
   const values = useMemo(() => [value], [value]);
 
-  const [state] = useApplicationState();
-
+  const [state, dispatch] = useApplicationState();
   const [swatchLayout, setSwatchLayout] = useState<SwatchLayout>('grid');
 
   const sharedSwatches = Selectors.getSharedSwatches(state);
 
+  const isSwatch =
+    value.swatchID &&
+    sharedSwatches.some((e) => e.do_objectID === value.swatchID);
```

Comment:
I think theoretically we shouldn't need to check `sharedSwatches` at all, cause all `swatchID` should be valid if they exist? Did you find this was needed?

Reply:
Maybe if we delete a swatch but they color still has the swatchId? 🤔

Reply:
In that case, we should probably update our swatch deletion code to traverse layers/styles and delete the swatchIDs. Similar to how we do `visitLayerColors` and `visitStyleColors` with "setSwatchColor"

Reply:
Do you want to update the deletion code in this PR? Or should we do it another time?

Reply:
Now that we've updated the swatch deletion, can we delete this `.some` check?